packages/dev/core/src/AudioV2/abstractAudio/abstractSound.ts

```diff
@@ -0,0 +1,308 @@
+import { Observable } from "../../Misc/observable";
+import type { Nullable } from "../../types";
+import { SoundState } from "../soundState";
+import { _AudioNodeType, AbstractNamedAudioNode } from "./abstractAudioNode";
+import type { _AbstractSoundInstance } from "./abstractSoundInstance";
+import type { PrimaryAudioBus } from "./audioBus";
+import type { AudioEngineV2 } from "./audioEngineV2";
+import type { _AbstractAudioSubGraph } from "./subNodes/abstractAudioSubGraph";
+import { _AudioSubNode } from "./subNodes/audioSubNode";
+import type { _VolumeAudioSubNode, IVolumeAudioOptions } from "./subNodes/volumeAudioSubNode";
+import { _VolumeAudioDefaults } from "./subNodes/volumeAudioSubNode";
+import type { AbstractSpatialAudio, ISpatialAudioOptions } from "./subProperties/abstractSpatialAudio";
+import type { AbstractStereoAudio, IStereoAudioOptions } from "./subProperties/abstractStereoAudio";
+
+/**
+ * Options for playing a sound.
+ */
+export interface ICommonSoundPlayOptions extends IVolumeAudioOptions {
+    /**
+     * Whether the sound should loop. Defaults to `false`.
+     */
+    loop: boolean;
+    /**
+     * The time within the sound buffer to start playing at, in seconds. Defaults to `0`.
+     */
+    startOffset: number;
+}
+
+/**
+ * Options for creating a sound.
+ */
+export interface ICommonSoundOptions extends ICommonSoundPlayOptions, ISpatialAudioOptions, IStereoAudioOptions {
+    /**
+     * Whether the sound should start playing immediately. Defaults to `false`.
+     */
+    autoplay: boolean;
+    /**
+     * The maximum number of instances that can play at the same time. Defaults to `Infinity`.
+     */
+    maxInstances: number;
+    /**
+     * The output bus for the sound. Defaults to the audio engine's default main bus.
+     * @see {@link AudioEngineV2.defaultMainBus}
+     */
+    outBus: PrimaryAudioBus;
+}
+
+/**
+ * Abstract class representing a sound in the audio engine.
+ */
+export abstract class AbstractSound extends AbstractNamedAudioNode {
+    private _outBus: Nullable<PrimaryAudioBus> = null;
+    private _state: SoundState = SoundState.Stopped;
+
+    protected _instances = new Set<_AbstractSoundInstance>();
+    protected _options = {} as ICommonSoundOptions;
+    protected abstract _subGraph: _AbstractAudioSubGraph;
+
+    /**
+     * Observable for when the sound stops playing.
+     */
+    public readonly onEndedObservable = new Observable<AbstractSound>();
+
+    protected constructor(name: string, engine: AudioEngineV2, options: Partial<ICommonSoundOptions> = {}) {
+        super(name, engine, _AudioNodeType.Out);
+
+        Object.assign(this._options, options);
+        this._options.autoplay ??= false;
+        this._options.loop ??= false;
+        this._options.maxInstances ??= Infinity;
+        this._options.startOffset ??= 0;
+        this._options.volume ??= _VolumeAudioDefaults.Volume;
```

Comment:
```suggestion
        this._options = {
           autoplay: false,
           loop: false,
           maxInstances: Infinity,
           startOffset: 0,
           volume: _VolumeAudioDefaults.Volume,
           ...options
       }
```

Also no need to create it in line 56.

This avoids a lot of `??=` which in any version under es2020 (i.e. UMD) will generate a lot of code.